### PR TITLE
Convert project to Gradle

### LINKED ISSUE #3351: Unable to build project using mvn 3.6.0, java 11
Unable to build project using maven 3.6.0 & java 11

```
mvn -DskipTests clean compile package 
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$1 (file:/usr/share/maven/lib/guice.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for com.squareup.retrofit2:retrofit:jar:2.9.0-SNAPSHOT
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-jar-plugin is missing. @ com.squareup.retrofit2:retrofit:[unknown-version], /home/pi/retrofit/retrofit/pom.xml, line 98, column 15
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] 
[INFO] ------------------< com.squareup.retrofit2:retrofit >-------------------
[INFO] Building Retrofit 2.9.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ retrofit ---
[INFO] Deleting /home/pi/retrofit/retrofit/target
[INFO] 
[INFO] --- maven-enforcer-plugin:1.0:enforce (enforce-maven) @ retrofit ---
[INFO] 
[INFO] --- kotlin-maven-plugin:1.3.50:compile (compile) @ retrofit ---
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ retrofit ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.6.1:compile (default-compile) @ retrofit ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 50 source files to /home/pi/retrofit/retrofit/target/classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/pi/retrofit/retrofit/src/main/java/retrofit2/Utils.java:[18,15] cannot access java.io.IOException
  bad class file: /modules/java.base/java/io/IOException.class
    class file has wrong version 55.0, should be 53.0
    Please remove or make sure it appears in the correct subdirectory of the classpath.
[INFO] 1 error
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  30.986 s
[INFO] Finished at: 2020-04-04T20:44:02+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.1:compile (default-compile) on project retrofit: Compilation failure
[ERROR] /home/pi/retrofit/retrofit/src/main/java/retrofit2/Utils.java:[18,15] cannot access java.io.IOException
[ERROR]   bad class file: /modules/java.base/java/io/IOException.class
[ERROR]     class file has wrong version 55.0, should be 53.0
[ERROR]     Please remove or make sure it appears in the correct subdirectory of the classpath.
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

```
pi@raspberrypi:~/retrofit/retrofit $ mvn --version
Apache Maven 3.6.0
Maven home: /usr/share/maven
Java version: 11.0.6, vendor: Raspbian, runtime: /usr/lib/jvm/java-11-openjdk-armhf
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.19.97-v7l+", arch: "arm", family: "unix"
```

```
pi@raspberrypi:~/retrofit/retrofit $ java -version
openjdk version "11.0.6" 2020-01-14
OpenJDK Runtime Environment (build 11.0.6+10-post-Raspbian-1deb10u1)
OpenJDK Server VM (build 11.0.6+10-post-Raspbian-1deb10u1, mixed mode)
```

### PR DESCRIPTION
**Pull request by JakeWharton**
This allows integrating an Android test project easily.

Closes #3351.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+POM_SCM_DEV_CONNECTION=scm:git:ssh://git@github.com/square/retrofit.git
+
+POM_LICENCE_NAME=The Apache Software License, Version 2.0
+POM_LICENCE_URL=http://www.apache.org/licenses/LICENSE-2.0.txt
+POM_LICENCE_DIST=repo
+
+POM_DEVELOPER_ID=square
+POM_DEVELOPER_NAME=Square, Inc.
+
+# Publishing SHA 256 and 512 hashses of maven-metadata is not supported by Sonatype and Nexus.
```
Comment (Egorand) [inline gradle.properties:16]: nit: `hashses`

Comment (JakeWharton) [reply gradle.properties:16]: fixed

File: retrofit-converters/wire/build.gradle [hunk 2]
```diff
@@ -0,0 +1,21 @@
+apply plugin: 'java-library'
+apply plugin: 'com.vanniktech.maven.publish'
+
+dependencies {
+  api project(':retrofit')
+  api(deps.wireRuntime) {
+    // Make sure OkHttp's transitive version wins (itself transitive from Retrofit).
```
Comment (swankjesse) [inline retrofit-converters/wire/build.gradle:7]: Why’s this necessary? I’d expect the highest number would be better policy

File: retrofit/build.gradle [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  testImplementation deps.guava
+  testImplementation deps.mockwebserver
+  testImplementation deps.robolectric
+  testImplementation deps.kotlinStdLib
+  testImplementation deps.kotlinCoroutines
+  testImplementation deps.android
+}
+
+test {
+  // Robolectric (at least the version we use) only works on JDK 8.
```
Comment (swankjesse) [inline retrofit/build.gradle:28]: Do these crash before setUp? I’d prefer assume failures over excluding tests ’cause it’s more visible when they don’t run.

File: retrofit-converters/wire/build.gradle (hunk 2 above)
Comment (JakeWharton) [reply retrofit-converters/wire/build.gradle:7]: Unfortunately Maven isn't that smart. Okio from Retrofit is a 3rd order dependency and Okio from Wire is a 2nd order dep. So Maven selects the Wire one.

Gradle consumers don't have this problem.

File: retrofit/build.gradle (hunk 3 above)
Comment (JakeWharton) [reply retrofit/build.gradle:28]: Yes. In the runner creation. Robolectric goes away in two PRs after this one so not a huge deal.

**Pull request merged**